src/include/Context.h

```diff
@@ -463,6 +464,19 @@ class FunctionContext : public Context {
   boost::function<void(int)> m_callback;
 };
 
+class LambdaContext : public Context {
```

Comment:
There isn't a formulated plan for this in particular, but there has been talk about switching to C++11 primitives where available.  Such cleanup work should probably be left until the kraken release at this point.